client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -364,6 +365,14 @@
      */
     CompletableFuture<Transaction.Status> checkTransactionStatus(final Stream stream, final UUID txId);
 
+    /**
+     * Gets an {@link AsyncIterator} on TxnResponse in stream.
+     *
+     * @param stream The name of the stream for which to list completed transaction.
+     * @return An {@link AsyncIterator} which can be used to iterate over all TxnResponse in the stream.
+     */
+    AsyncIterator<TxnResponse> listCompletedTransactions(final Stream stream);
+
```

Comment:
This is abstraction breaking. TxnResponse is a grpc structure not a client data structure. All the other calls on this interface return a type which is meaningful to the client and hide the internals in the implementation. Perhaps this should be a Transaction or a TransactionId.

Reply:
Created new interface named as **TransactionInfo** having detail about stream, transactionId and transactionStatus . Replaced grpc structure i.e TxnResponse with **TransactionInfo**.